torch_xla/csrc/tensor.cpp

```diff
@@ -1114,9 +1114,18 @@ XLATensor XLATensor::cat(tensorflow::gtl::ArraySlice<const XLATensor> tensors,
                          xla::int64 dim) {
   XLA_CHECK_GT(tensors.size(), 0);
   std::vector<ir::Value> values;
+
+  bool set_dim = false;
```

Comment:
I'd prefer simpler code and not keeping the flag around.
When dim turns positive, further calls to GetCanonicalDimension() are just noops.